algorithm-registry.md

```diff
@@ -0,0 +1,64 @@
+# Algorithm Registry
+
+This file is designed to act as a source of truth regarding what signing
+algorithms are recommended across the Sigstore ecosystem. Any changes to this
+file **must** be reflected in the `PublicKeyDetails` enumeration in
+[`sigstore_common.proto`] in [sigstore/protobuf-specs].
+
+Sigstore clients aren't required to support all algorithms in this registry,
```

Comment:
Can we file an issue to track figuring out what set of algorithms clients are _required_ to support? Probably ecdsa p256 (for ephemeral sigs), ecdsa p384 (for root key material) and ed25519 (for witness cosigs).

Reply:
Yeah! I'll file that.